src/tty.rs

```diff
@@ -203,17 +204,44 @@ fn get_pw_entry(buf: &mut [i8; 1024]) -> Passwd {
 }
 
 /// Exec a shell
-fn execsh(config: &Config) -> ! {
+fn execsh(config: &Config, options: &Options) -> ! {
     let mut buf = [0; 1024];
     let pw = get_pw_entry(&mut buf);
 
-    let shell = match config.shell() {
-        Some(shell) => match shell.to_str() {
-            Some(shell) => shell,
-            None => die!("Invalid shell value")
-        },
-        None => pw.shell
-    };
+    // We construct a string with the shell path and an array of C-string
+    // pointers to pass to execv.
+    // Shells are searched in the following order:
+    //   Command line argument
+    //   Alacritty's Configuration
+    //   User's shell from passwd
+    // Config _cannot_ return a reference, therefore it has to be stored in
+    // this scope.
+    let argv_conf = config.shell();
+    let shell;
+    let mut argv = Vec::new();
+    if let Some(ref argv_opt) = options.shell {
```

Comment:
I dont think CString is serilizable and we have to give cstrings to execv.. So the array of cstrings have to be owned in this scope